### PR TITLE
Mount Propagation support for Nomad managed Volumes

### PR DESCRIPTION
**Pull request by endocrimes**

### CODE REVIEW

File: api/tasks.go [hunk 1]
```diff
@@ -374,9 +374,10 @@ type VolumeRequest struct {
 // VolumeMount represents the relationship between a destination path in a task
```
Comment (preetapan) [inline api/tasks.go]: Would need docs both for volume/volume_mount stanza and potentially mentioning it in the docker driver docs linking back to the stanza

File: drivers/shared/executor/executor_linux.go [hunk 2]
```diff
@@ -821,6 +821,19 @@ func cmdDevices(devices []*drivers.DeviceConfig) ([]*lconfigs.Device, error) {
...
 }
 
+var mountPropagationMapping = map[string]int{
+	"private": unix.MS_PRIVATE,
+	"rslave":  unix.MS_SLAVE | unix.MS_REC,
+	"rshared": unix.MS_SHARED | unix.MS_REC,
+}
+
+var userMountToUnixMount = map[string]string{
+	"":              "private",
```
Comment (nickethier) [inline drivers/shared/executor/executor_linux.go]: could we use some consts for these?

File: drivers/docker/driver.go [hunk 3]
```diff
@@ -840,13 +847,24 @@ func (d *Driver) createContainerConfig(task *drivers.TaskConfig, driverConfig *T
...
 			Source:   m.HostPath,
 			ReadOnly: m.Readonly,
-		})
+		}
+
+		// MountPropagation is only supported by Docker on Linux:
+		// https://docs.docker.com/storage/bind-mounts/#configure-bind-propagation
+		if runtime.GOOS == "linux" {
+			hm.BindOptions = &docker.BindOptions{
+				Propagation: userMountToUnixMount[m.PropagationMode],
```
Comment (nickethier) [inline drivers/docker/driver.go]: bidirectional/rshared is kind of dangerous. Should we limit it's use in anyway?

**Comment by endocrimes** [comment]
Closing in favour of new, follow up PR.

**Pull request closed without merging**